### PR TITLE
feat: new use case for benchmarking influxdb 1.x vs 2.x metaquery performance

### PR DESCRIPTION
**Pull request by wbaker85**
This PR creates a new use case designed to allow us to generate data & benchmark metaquery performance for influxdb 1.x and influxdb 2.x for the purposes of comparison and optimization of 2.x metaqueries.

The use case is more abstract than the existing DevOps, IoT etc. use cases in that it essentially allows us to create a specified amount of cardinality in the form of many different tag values for two constant tag keys over a specified time range. This is intended to emulate specific customer issues with metaquery performance differences between the two influxdb versions, and does a great job at highlighting the difference in performance.

The plan is to use this use case as part of our automated performance testing for influxdb.

### CODE REVIEW

File: bulk_data_gen/metaqueries/generate_data.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	"time"
+
+	"math/rand"
+
+	"github.com/influxdata/influxdb-comparisons/bulk_data_gen/common"
+)
+
+// letters are used for generating the pseudo-random tag values used in this
+// dataset. Currently only the first 10 letters are used.
+const letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
```
Comment (codyshepherd) [inline bulk_data_gen/metaqueries/generate_data.go:13]: Just out of curiosity, why not use something like a truncated [Type 4 UUID](https://pkg.go.dev/github.com/google/uuid#NewRandom) instead?

**Review by mkm-influx — APPROVED** [review]

File: bulk_data_gen/metaqueries/generate_data.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	axis    int
+	TagList map[int][]byte
+
+	timestampNow   time.Time
+	timestampStart time.Time
+	timestampEnd   time.Time
+	stepTime       time.Duration
+}
+
+func (g *MetaquerySimulator) SeenPoints() int64 {
```
Comment (codyshepherd) [inline bulk_data_gen/metaqueries/generate_data.go:80]: Any particular reason the names of these functions don't match the member names or aren't named as getters?

File: bulk_query_gen/influxdb/influx_metaquery_common.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+// query is slightly more complicated and is intended to replicate an equivalent
+// query that would be generated by the InfluxDB UI for getting "all" the tag
+// values for a specific tag key.
+func (d *InfluxMetaquery) StandardMetaquery(qi bulkQuerygen.Query) {
+	var query string
+	if d.language == InfluxQL {
+		query = `SHOW TAG VALUES FROM "example_measurement" WITH KEY = "X" LIMIT 200`
+	} else {
+		query = fmt.Sprintf(`from(bucket: "%s") `+
+			`|> range(start: %s, stop: %s) `+
```
Comment (codyshepherd) [inline bulk_query_gen/influxdb/influx_metaquery_common.go:42]: You might want someone with more expertise in flux than me take a look at this query.

File: bulk_query_gen/metaquery.go [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package bulk_query_gen
+
+type Metaquery interface {
+	StandardMetaquery(Query)
+	Dispatch(int) Query
+}
+
+func MetaqueryDispatchAll(d Metaquery, iteration int, q Query, scaleVar int) {
```
Comment (codyshepherd) [inline bulk_query_gen/metaquery.go:8]: Why bother with the iteration and scaleVar arguments when they're not used?

**Review by codyshepherd — APPROVED** [review]
A few questions and comments but no real blockers from what I can see.

File: bulk_query_gen/influxdb/influx_metaquery_common.go (hunk 3 above)
Comment (danxmoran) [reply bulk_query_gen/influxdb/influx_metaquery_common.go:42]: This LGTM, `keep(columns:[tag]) |> group() |> distinct(column: tag)` is the pattern that we'd expect to trigger the Flux pushdown

File: bulk_data_gen/metaqueries/generate_data.go (hunk 2 above)
Comment (wbaker85) [reply bulk_data_gen/metaqueries/generate_data.go:80]: There's an existing interface `Simulator` that it needs to satisfy: https://github.com/influxdata/influxdb-comparisons/blob/master/bulk_data_gen/common/simulation.go#L17-L23

I'll add a comment about that so it's more clear.

File: bulk_query_gen/influxdb/influx_metaquery_common.go (hunk 3 above)
Comment (wbaker85) [reply bulk_query_gen/influxdb/influx_metaquery_common.go:42]: It's based on the queries that the UI uses for getting metadata, so I think it should be representative.

File: bulk_data_gen/metaqueries/generate_data.go (hunk 1 above)
Comment (wbaker85) [reply bulk_data_gen/metaqueries/generate_data.go:13]: Good idea! I'll incorporate that. Looks like it makes the data generation slightly more efficient.

File: bulk_query_gen/metaquery.go (hunk 4 above)
Comment (wbaker85) [reply bulk_query_gen/metaquery.go]: I was basically using other similar functions as a template, but yeah there's not any reason to have those extra args, so I'll remove them!

**Review by codyshepherd — APPROVED** [review]

**Pull request merged**